EspionSpotify.Updater/Program.cs

```diff
@@ -0,0 +1,137 @@
+        {
+            Console.WriteLine("Downloading update...");
+            if (!downloadUpdate())
+            {
+                Console.WriteLine("Press any key to exit");
+                Console.ReadKey();
+                Environment.Exit(1);
+            }
+            try
+            {
+                Console.WriteLine("Extracting");
+                extract(AppDomain.CurrentDomain.BaseDirectory);
+            }
+            catch (System.UnauthorizedAccessException ex)
+            {
+                Console.WriteLine("An error occurred while extracting. Make sure Spytify is not running or Antivirus Software is interferring.\n"+
+                    "You can extract the file manually at: "+ AppDomain.CurrentDomain.BaseDirectory+"../update.zip\n"+
+                    "Error Message: "+ ex.Message);
+                Console.WriteLine("Press any key to exit");
+                Console.ReadKey();
+                Environment.Exit(1);
+            }
+            catch(Exception ex)
+            {
+                Console.WriteLine("An error occurred while extracting\n" +
+                    "The error was" + ex.Message);
+                Console.ReadKey();
+            }
+            File.Delete(AppDomain.CurrentDomain.BaseDirectory + "../update.zip");
+            Console.WriteLine("Update successfull");
+            Process.Start(new ProcessStartInfo(AppDomain.CurrentDomain.BaseDirectory + "../Spytify.exe", "upgradeSettings"));
+            Environment.Exit(0);
+        }
+        public static bool isVersionNewer(string name)
+        {
+            Match m = Regex.Match(name, @"v\.((?:\d+\.)*\d+?).zip$");
+            if (m.Groups[1].Success)
+            {
+                string currentVersion = null;
+                try
+                {
+                    currentVersion = FileVersionInfo.GetVersionInfo(AppDomain.CurrentDomain.BaseDirectory + "../Spytify.exe").FileVersion;
+                }
+                catch
+                {
+                    return true;
+                }
+                
+                if (currentVersion.StartsWith(m.Groups[1].Value))
+                    return false;
+                return true;
+            }
+            return true;
+        }
+        public static bool downloadUpdate()
+        {
+            try
+            {
+                HttpWebRequest wr = (HttpWebRequest)WebRequest.Create("https://api.github.com/repos/jwallet/spy-spotify/releases/latest");
+                wr.Method = WebRequestMethods.Http.Get;
+                wr.UserAgent = "Spytify";
+                WebResponse ws = wr.GetResponse();
+                string json = new StreamReader(ws.GetResponseStream()).ReadToEnd();
+                Release release = JsonConvert.DeserializeObject<Release>(json);
+                Asset targetRelease = null;
+                foreach (Asset asset in release.assets)
+                {
+                    if (Regex.IsMatch(asset.name, @"v\.((?:\d+\.)*\d+?).zip$"))
```

Comment:
official download will always start with spytify and end with zip "Spytify-v1.8.1.zip", so we want to ignore those:
`ADVERTISEMENT-ONLY-RECORDER-VERSION.-Spytify-v.1.8.zip`
`Spytify-v.1.8-debug.zip`
and version will always have major, but other are optional, so what I mean is:
`v.MAJOR.MINOR.PATCH` -- 4th number BUILD is always ignored, so it won't appear
so maybe a strict regex test will be better

```
var r = new System.Text.RegularExpressions.Regex(@"^Spytify(-|\.)(v?)(\.)?\d+(\.\d+)?(\.\d+)?.zip$", System.Text.RegularExpressions.RegexOptions.IgnoreCase);

Console.WriteLine(r.IsMatch("ADVERTISEMENT-ONLY-RECORDER-VERSION.-Spytify-v.1.8.zip")); // false
Console.WriteLine(r.IsMatch("Spytify-v.1.8-debug.zip")); // false
Console.WriteLine(r.IsMatch("Spytify.va.b.zip")); // false

Console.WriteLine(r.IsMatch("Spytify-v.1.zip")); // true - major
Console.WriteLine(r.IsMatch("Spytify-v.1.8.zip")); // true - major minor
Console.WriteLine(r.IsMatch("Spytify-v.1.8.1.zip")); // true - major minor patch

Console.WriteLine(r.IsMatch("spytify-v.1.zip")); // true - lowercase

Console.WriteLine(r.IsMatch("Spytify.v1.zip")); // true - with dot version

Console.WriteLine(r.IsMatch("Spytify-v1.zip")); // true - with dash version no-dot
```

Reply:
Ok
I changed the regex now to the one you suggested